### PR TITLE
docs: add initial documentation

### PR DESCRIPTION
**Pull request by grvcoelho**
Add initial documentation covering Install,  Usage and Steps configuration.

### CODE REVIEW

**Comment by grvcoelho** [comment]
This closes #3

File: README.md [hunk 1]
```diff
@@ -1,3 +1,144 @@
...
+
+## Deploy
+
+Available types are: `lambda` and `nomad`.
+
+### Lambda
+
+The Lambda Deploy step updates and deploys existing AWS Lambda functions. It creates a zip file, uploads it to an S3 Bucket and updates the Lambda functions code. 
+
+There's a convention that must be followed to name the Lambda functions. The functions names must have the following shape: `env-projectName-functionName`. So a function called `hello-world` from the `pagarme` project on the `live` environment must be called `live-pagarme-hello-world.
```
Comment (mccraveiro) [inline README.md]: Why the change in the convention? We're using `projectName-stage-functionName` before.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,142 @@
...
+  image_repository: xxxxxx.dkr.ecr.us-east-1.amazonaws.com/deployer # Pushing the image to an AWS ECR
+```
+
+## Deploy
+
+Available types are: `lambda` and `nomad`.
+
+### Lambda
+
+The Lambda Deploy step updates and deploys existing AWS Lambda functions. It creates a zip file, uploads it to an S3 Bucket and updates the Lambda functions code.
```
Comment (felquis) [inline README.md]:
```
Uploads it to a S3 Bucket 
```

File: README.md (hunk 1 above)
Comment (grvcoelho) [inline README.md]: This convention is more consistent with other parts of our infrastructure. Things are commonly named `stage-project-xxx`

The `project-stage-xxx` format was used by the Serverless Framework, but even Serverless didn't follow a proper convention: it uses `project-stage-xxx` for Lambda functions, but `stage-project-xxx` to API Gateway.

File: README.md (hunk 2 above)
Comment (grvcoelho) [inline README.md]: Fixed it :+1:

**Comment by felquis** [comment]
Next time you should consider rewrite the commits before merging 

screenshot_20170628-115355

**Comment by grvcoelho** [comment]
@felquis: you're absolutelly right! I've missed that!
